File: examples/10_instancing/asset/script/main.lua
local model = sn.Model.new()
model:load("DamagedHelmet.glb")
local material = model:getMaterial()

local transforms = {}
for i = 0, 10 do
    for j = 0, 10 do
        local t = sn.Transform.new()
        t.position = sn.Vec3.new(i * 2, 0, j * 2)
        t.rotation = sn.Vec3.new(90, 0, 180)
        table.insert(transforms, t)
    end
end

local pos = sn.Vec3.new(-3, 5, -3)
local at = sn.Vec3.new(15, -5, 15)
local up = sn.Vec3.new(0, 1, 0)
sn.Graphics.getCamera():lookat(pos, at, up)
sn.Graphics.bindPipeline(sn.BuiltinPipelines.get3DInstanced())

function Update()
    if sn.Keyboard.isPressed(sn.Keyboard.ESCAPE) then
        sn.Script.load("main", ".")
    end
end

function Draw()
    sn.Graphics.drawModelInstanced(model, transforms, material)
end

File: examples/10_instancing/asset/script/main.py
import sinen as sn
import math

model = sn.Model()
model.load("DamagedHelmet.glb")
material = model.get_material()

transforms = []
for i in range(10):
    for j in range(10):
        t = sn.Transform()
        t.position = sn.Vec3(i * 2, 0, j * 2)
        transforms.append(t)


pos = sn.Vec3(-3, 5, -3)
at = sn.Vec3(15, -5, 15)
up = sn.Vec3(0, 1, 0)
sn.Graphics.get_camera().lookat(pos, at, up)

sn.Graphics.bind_pipeline(sn.BuiltinPipelines.get_3d())


def update():
    sn.Logger.info("hello")
    if sn.Keyboard.is_pressed(sn.Keyboard.ESCAPE):
        sn.Script.load("main", ".")


def draw():
    sn.Graphics.draw_model_instanced(model, transforms, material)
